File: sportzone/backend/src/main/java/com/yourorg/appname/service/ProductService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.response.ProductDetailResponse;
import com.yourorg.appname.dto.response.ProductResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;
import java.util.List;

public interface ProductService {
    Page<ProductResponse> getProducts(String category, String brand, BigDecimal minPrice, BigDecimal maxPrice, Boolean inStock, String query, Pageable pageable);
    ProductDetailResponse getProductById(Long id);
    ProductDetailResponse getProductBySlug(String slug);
    List<ProductResponse> getFeaturedProducts();
    List<ProductResponse> getNewArrivals();
}

File: frontend/src/services/productService.js
import apiClient from './apiClient';
import { API_ENDPOINTS } from '../constants/apiEndpoints';

export const productService = {
  getProducts: async (params = {}) => {
    const response = await apiClient.get(API_ENDPOINTS.PRODUCTS.LIST, { params });
    return response.data;
  },

  getProductById: async (id) => {
    const response = await apiClient.get(API_ENDPOINTS.PRODUCTS.DETAIL(id));
    return response.data;
  },

  getProductBySlug: async (slug) => {
    const response = await apiClient.get(API_ENDPOINTS.PRODUCTS.BY_SLUG(slug));
    return response.data;
  },

  getFeaturedProducts: async () => {
    const response = await apiClient.get(API_ENDPOINTS.PRODUCTS.FEATURED);
    return response.data;
  },

  getNewArrivals: async () => {
    const response = await apiClient.get(API_ENDPOINTS.PRODUCTS.NEW_ARRIVALS);
    return response.data;
  },
};
